### PR TITLE
chore: set app setting config required for all all 3 service uris

### LINKED ISSUE #3207: [AVM Module Issue]: DurableTask.AzureStorage: If at least one Azure Storage service URI is specified, BlobServiceUri, QueueServiceUri, and TableServiceUri must all be provided.
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Module Name

avm/res/web/site

### (Optional) Module Version

_No response_

### Description

When using this AVM to deploy a function app and when using 'storageAccountUseIdentityAuthentication: true', it automatically creates 2x app settings: AzureWebJobsStorage__accountName and AzureWebJobsStorage__blobServiceUri.

This works fine when deploying standard (non-durable) functions to the function app, however when you deploy a durable function, the runtime presents the below error:

DurableTask.AzureStorage: If at least one Azure Storage service URI is specified, BlobServiceUri, QueueServiceUri, and TableServiceUri must all be provided.

If AzureWebJobsStorage__blobServiceUri is removed, the runtime succeeds.

This can be worked around by adding the queue/table endpoints to the provided parameters, but I would prefer to see this logic added to the AVM.

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by pankajagrawal16**
## Description

Closes #3207 

## Pipeline Reference

| Pipeline |
| -------- |
|  [avm.res.web.site](https://github.com/pankajagrawal16/bicep-registry-modules/actions/workflows/avm.res.web.site.yml)   |

## Type of Change

- [x] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [x] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/web/site/config--appsettings/main.bicep
Comment (AlexanderSehr) [inline avm/res/web/site/config--appsettings/main.bicep:1]: Please note, the `main.json` templates **must** be re-build with the latest Bicep version `0.31.34` as this version (running on the agents) renders a different `main.json` file then previous versions. 

The steps to execute would be
- Upgrade your Bicep CLI version to `0.31.34` (e.g. via `choco upgrade bicep` if choco is used)
- Re-run the `Set-AVMModule` script on your module. E.g. via `Set-AVMModule -ModuleFolderPath '.\avm\res\automation\automation-account\' -SkipFileAndFolderSetup -SkipReadMe -Recurse`
- Push the changes
- Re-run the tests to ensure everything works. 

If the static tests continue to fail for you, please note that there is currently presumably another new bug which affects some of our modules ([ref](https://github.com/Azure/bicep/issues/15517)).

**Comment by jtracey93** [comment]
Hey @pankajagrawal16 ,

Firstly, thanks for your work on this PR!

We have made some changes to the AVM CI, detailed below, which means we need you to update your fork to pull in these latest changes and re-run your tests to show they still are passing prior to approving and merging this PR, as we don't and it fails once merged the publishing of your module will fail and will be blocked going forward until the test pass again via additional PRs.

### Changes to CI That Have Been Made That You Need To Take Action On

- https://github.com/Azure/bicep-registry-modules/pull/3745
    - Adds WAF Security PSRule Checks using this custom baseline of rules ([`/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/cb-waf-security.Rule.yaml`](https://github.com/Azure/bicep-registry-modules/blob/main/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/cb-waf-security.Rule.yaml))
    - **Action Required By You:**
        - Ingest/pull-in these changes into your PR branch and re-run your tests to ensure they pass
            - If any issues with the rules and you think it should apply, please ping/reach out to the core team

Any questions reach out to the AVM Core Team by tagging us in your PR here or internally via Teams

Thanks

Jack (AVM Core Team)

File: avm/res/web/site/config--appsettings/main.bicep
Comment (pankajagrawal16) [reply avm/res/web/site/config--appsettings/main.bicep:1]: let me check on this.

Comment (pankajagrawal16) [reply avm/res/web/site/config--appsettings/main.bicep:1]: @AlexanderSehr Does it look better ? I updated the bicep cli to latest and ran the script again

**Comment by pankajagrawal16** [comment]
@Azure/avm-core-team  So i Pulled latest changes here and some static validations failed like this https://github.com/pankajagrawal16/bicep-registry-modules/actions/runs/12069251366/job/33656136154

I do not think its valid failure since the module itself sets tls version 1.2 by default. 

What do you think should be done for such cases ?

**Comment by AlexanderSehr** [comment]
> @Azure/avm-core-team So i Pulled latest changes here and some static validations failed like this https://github.com/pankajagrawal16/bicep-registry-modules/actions/runs/12069251366/job/33656136154
> 
> I do not think its valid failure since the module itself sets tls version 1.2 by default. Same goes for ftpsState state
> 
> What do you think should be done for such cases ?

Hey @pankajagrawal16,
just took a brief look and there are 'some' issues. For example, when looking at the test `winContainerWebApp.defaults` which is one that fails the TLS test. I can see that the default value of the site config does set the version to 1.2 by default - but the test then overwrites that default as it provides a value for this parameter:
```biep
siteConfig: {
  appSettings: [
    {
      name: 'WEBSITES_ENABLE_APP_SERVICE_STORAGE'
      value: 'false'
    }
  ]
  windowsFxVersion: 'DOCKER|mcr.microsoft.com/azure-app-service/windows/parkingpage:latest'
}
```

So a fix in this case 'should' for example be to add `minTlsVersion: '1.2'` to that object.

On the other end of the spectrum I find it a quite odd that there's a failing test that **demands** the use of an MSI - even if no none is needed. For example: 
image

It should be possible to disable this test in the PSRule configuration, though I don't think we had the case yet for the PSRule security tests. @eriqua I guess we'd need to introduce a new configuration file, right? Mind you though @pankajagrawal16 disabling tests should only seen as the last resort if a test fundamentally does not seem to make sense.

**Comment by eriqua** [comment]
> > @Azure/avm-core-team So i Pulled latest changes here and some static validations failed like this https://github.com/pankajagrawal16/bicep-registry-modules/actions/runs/12069251366/job/33656136154
> > I do not think its valid failure since the module itself sets tls version 1.2 by default. Same goes for ftpsState state
> > What do you think should be done for such cases ?
> 
> Hey @pankajagrawal16, just took a brief look and there are 'some' issues. For example, when looking at the test `winContainerWebApp.defaults` which is one that fails the TLS test. I can see that the default value of the site config does set the version to 1.2 by default - but the test then overwrites that default as it provides a value for this parameter:
> 
> ```
> siteConfig: {
>   appSettings: [
>     {
>       name: 'WEBSITES_ENABLE_APP_SERVICE_STORAGE'
>       value: 'false'
>     }
>   ]
>   windowsFxVersion: 'DOCKER|mcr.microsoft.com/azure-app-service/windows/parkingpage:latest'
> }
> ```
> 
> So a fix in this case 'should' for example be to add `minTlsVersion: '1.2'` to that object.
> 
> On the other end of the spectrum I find it a quite odd that there's a failing test that **demands** the use of an MSI - even if no none is needed. For example: image
> 
> It should be possible to disable this test in the PSRule configuration, though I don't think we had the case yet for the PSRule security tests. @eriqua I guess we'd need to introduce a new configuration file, right? Mind you though @pankajagrawal16 disabling tests should only seen as the last resort if a test fundamentally does not seem to make sense.

Hey @AlexanderSehr, if the decision is to exempt the rules, I don't think a new config file would be needed. As far as I see the failing rules are only for the custom baseline and only for default tests. For defaults tests we already have the min suppression group config file.

Regarding what I see above, by checking the latest run, I see the blocking failed rules are Azure.AppService.WebSecureFtp and the min tls one. The managed identity should be a security one, but not custom baseline one, hence not blocking.

**Comment by pankajagrawal16** [comment]
@AlexanderSehr Rite, i missed the site config override part. I will fix that in tests.

**Comment by AlexanderSehr** [comment]
> > > @Azure/avm-core-team So i Pulled latest changes here and some static validations failed like this https://github.com/pankajagrawal16/bicep-registry-modules/actions/runs/12069251366/job/33656136154
> > > I do not think its valid failure since the module itself sets tls version 1.2 by default. Same goes for ftpsState state
> > > What do you think should be done for such cases ?
> > 
> > 
> > Hey @pankajagrawal16, just took a brief look and there are 'some' issues. For example, when looking at the test `winContainerWebApp.defaults` which is one that fails the TLS test. I can see that the default value of the site config does set the version to 1.2 by default - but the test then overwrites that default as it provides a value for this parameter:
> > ```
> > siteConfig: {
> >   appSettings: [
> >     {
> >       name: 'WEBSITES_ENABLE_APP_SERVICE_STORAGE'
> >       value: 'false'
> >     }
> >   ]
> >   windowsFxVersion: 'DOCKER|mcr.microsoft.com/azure-app-service/windows/parkingpage:latest'
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > So a fix in this case 'should' for example be to add `minTlsVersion: '1.2'` to that object.
> > On the other end of the spectrum I find it a quite odd that there's a failing test that **demands** the use of an MSI - even if no none is needed. For example: image
> > It should be possible to disable this test in the PSRule configuration, though I don't think we had the case yet for the PSRule security tests. @eriqua I guess we'd need to introduce a new configuration file, right? Mind you though @pankajagrawal16 disabling tests should only seen as the last resort if a test fundamentally does not seem to make sense.
> 
> Hey @AlexanderSehr, if the decision is to exempt the rules, I don't think a new config file would be needed. As far as I see the failing rules are only for the custom baseline and only for default tests. For defaults tests we already have the min suppression group config file.
> 
> Regarding what I see above, by checking the latest run, I see the blocking failed rules are Azure.AppService.WebSecureFtp and the min tls one. The managed identity should be a security one, but not custom baseline one, hence not blocking. In any case, if we wanted to get to a clean state and also exempt the managed identity one, although not blocking, we could still use the existing min config file. That is generic and should cover exemptions for any pillar/baseline

Thank you very much for the elaborate answer @eriqua. I may have mixed it up then. The new baseline still a blank spot for me 😄

**Comment by pankajagrawal16** [comment]
@tsc-buddy Can you have a Look at PR, all tests are passing now with latest changes pulled inn to as per discussion with Core team above.

File: avm/res/web/site/config--appsettings/main.bicep
Comment (AlexanderSehr) [reply avm/res/web/site/config--appsettings/main.bicep:1]: Perfect. Thanks @pankajagrawal16

**Review by tsc-buddy — APPROVED** [review]

**Pull request merged**
